sceptre/cli/dump.py

```diff
@@ -53,3 +55,55 @@ def dump_config(ctx, path):
     else:
         for config in output:
             write(config, output_format)
+
+
+@dump_group.command(name="template")
+@click.argument("path")
+@click.option(
+    "-n",
+    "--no-placeholders",
+    is_flag=True,
+    help="If True, no placeholder values will be supplied for resolvers that cannot be resolved.",
+)
+@click.pass_context
+@catch_exceptions
+def dump_template(ctx, no_placeholders, path):
+    """
+    Prints the template used for stack in PATH.
+    \f
+
+    :param path: Path to execute the command on.
+    :type path: str
+    """
+    context = SceptreContext(
+        command_path=path,
+        command_params=ctx.params,
+        project_path=ctx.obj.get("project_path"),
+        user_variables=ctx.obj.get("user_variables"),
+        options=ctx.obj.get("options"),
+        output_format=ctx.obj.get("output_format"),
+        ignore_dependencies=ctx.obj.get("ignore_dependencies"),
+    )
+
+    plan = SceptrePlan(context)
+
+    execution_context = (
+        null_context() if no_placeholders else use_resolver_placeholders_on_error()
+    )
+    with execution_context:
+        responses = plan.dump_template()
+
+    output = []
+    for stack, template in responses.items():
+        if template is None:
+            logger.warning(f"{stack.external_name} does not exist")
+        else:
+            output.append({stack.external_name: template})
```

Comment:
Why a list of dicts and not just one big dict?

Reply:
Again. For consistency with `dump config`. I can't remember to be honest why we did it this way.

Reply:
That feels a bit odd to me

Reply:
I _think_ it may be to work around or hack the limitations of the `write` helper. If we need to change `dump config`, perhaps this should be looked at in a separate issue/PR?

Reply:
My only concern is this: Let's say I need to pipe a template over to a linter or awscli for some reason because it has jinja and other stuff in it. I want to do something like: `sceptre dump template my/sceptre/config.yaml | cfn-lint -`. When you're outputting the templates inside a larger data structure like this, that's not a valid template. 

I understand your concern of wanting to be able to associate the template with a stack, so there are a few options I can think of:

Option 1: Rather than this, you could output the stack name via a logger rather than echoing it. That will result it going to stderr rather than stdout. That way, if you pipe output to another command or use `>`, it won't actually get included. So you could alternate by logging the stack name and then writing the output.

Option 2: Have a `--stack-names` flag that optionally makes it output this way but by default it does not.

Reply:
@jfalkenstein you are misunderstanding the implementation and behaviour. Here's a demo, showing a yaml stream that can be parsed by external tools is always emitted, even in the case of multiple stacks:

<img width="1770" alt="Screen Shot 2023-04-21 at 8 20 03 pm" src="https://user-images.githubusercontent.com/2305330/233612735-de6664aa-b83c-4bea-8a13-4bcbe6e33328.png">